File: src/terceiro-bimestre/pfe/s18/a2/Emprestimo.js
import Livro from './Livro.js';
import Usuario from './Usuario.js';

class Emprestimo {
  constructor(id, livro, usuario, data_emprestimo, data_devolucao) {
    this.__id = id;
    this.__livros = [];
    this.__livros.add_livro(livro);
    this.__usuario = usuario;
    this.__data_emprestimo = data_emprestimo;
    this.__data_devolucao = data_devolucao;
  }

  getId() {
    return this.__id;
  }

  getDataEmprestimo() {
    return this.__data_emprestimo;
  }

  getDataDevolucao() {
    return this.__data_devolucao;
  }

  add_livro(livro) {
    this.__livros.push(livro);
  }

  remover_livro(id) {
    let livro = this.busca_livro(id);
    if (livro) {
      this.__livros.pop(livro);
    } else {
      console.log(`Livro com o Id = ${id} não encontrado`);
    }
  }

  busca_livro(id) {
    return this.__livros.find((livro) => livro.getId() === id) || null;
  }

  mostrar_catalogo_livros() {
    this.__livros.forEach((livro) => {
      livro.getInformacoes();
      console.log("--------------------------------------");
    });
  }

  exibir_emprestimo() {
    console.log(`ID: ${this.getId()}`);
    this.mostrar_catalogo_livros();
    this.__usuario.getInformacoes();
    this.getDataEmprestimo();
    this.getDataDevolucao();
  }

}

File: src/quarto-bimestre/logica/biblioteca/Emprestimo.py
class Emprestimo:
    def __init__(self, id, usuario, livro, data_emprestimo, data_devolucao):
        self.id = id
        self.usuario = usuario
        self.livro = livro
        self.data_emprestimo = data_emprestimo
        self.data_devolucao = data_devolucao

    def get_id(self):
        return self.id
    
    def get_usuario(self):
        return self.usuario
    
    def get_livro(self):
        return self.livro
    
    def get_data_emprestimo(self):
        return self.data_emprestimo
    
    def get_data_devolucao(self):
        return self.data_devolucao

    def mostrar_emprestimo(self):
        print(f"\nID Empréstimo: {self.get_id()}")
        print(f"\nUsuário")
        self.get_usuario().mostra_usuario()
        print(f"\nLivro")
        self.get_livro().mostrar_livro()
        print(f"\nData de Empréstimo: {self.get_data_emprestimo()}")
        print(f"Data de Devolução: {self.get_data_devolucao()}")


